src/widgets/allergies/allergy-form.component.tsx

```diff
@@ -13,183 +13,169 @@ import {
-      const subscription = getAllergyReaction().subscribe(
-        data => setAllergyReaction(data),
+  useEffect(() => {
+    if (viewEditForm) {
+      const getAllergicReactionsSub = getAllergicReactions().subscribe(
+        (allergicReactions: Array<AllergicReaction>) =>
+          setAllergicReactions(allergicReactions),
         createErrorHandler()
       );
+      return () => {
+        getAllergicReactionsSub.unsubscribe();
+      };
+    }
+  }, [viewEditForm]);
+
+  useEffect(() => {
+    // allergenType is a required field in the Create form
+    if (!viewEditForm && allergenType) {
+      setEnableCreateButtons(true);
+    } else {
+      setEnableCreateButtons(false);
+    }
+  }, [allergenType, firstOnsetDate, viewEditForm]);
 
-      return () => subscription.unsubscribe();
+  useEffect(() => {
+    if (viewEditForm && formChanged) {
+      setEnableEditButtons(true);
+    } else {
+      setEnableEditButtons(false);
     }
-  }, [patientUuid, isLoadingPatient]);
+  }, [formChanged, viewEditForm]);
 
-  React.useEffect(() => {
-    if (selectedAllergyCategory) {
-      const sub1 = getAllergyAllergenByConceptUuid(
+  useEffect(() => {
+    if (selectedAllergyCategory && !viewEditForm) {
+      const getAllergensSub = getAllergyAllergenByConceptUuid(
         selectedAllergyCategory
-      ).subscribe(data => setAllergensArray(data), createErrorHandler);
-      const sub2 = getAllergyReaction().subscribe(
-        data => setAllergyReaction(data),
-        createErrorHandler
+      ).subscribe(data => setAllergensArray(data), createErrorHandler());
+      const getAllergicReactionsSub = getAllergicReactions().subscribe(
+        data => setAllergicReactions(data),
+        createErrorHandler()
       );
       return () => {
-        sub1.unsubscribe();
-        sub2.unsubscribe();
+        getAllergensSub.unsubscribe();
+        getAllergicReactionsSub.unsubscribe();
       };
     }
-  }, [selectedAllergyCategory]);
+  }, [selectedAllergyCategory, viewEditForm]);
 
-  const handleAllergenReactionChange = event => {
-    if (event.target.checked == true) {
-      setPatientAllergyReaction([
-        ...patientAlleryReaction,
-        { uuid: event.target.value }
-      ]);
+  const handleAllergicReactionChange = (
+    event: SyntheticEvent<HTMLInputElement, Event>
+  ) => {
+    if (event.currentTarget.checked === true) {
+      // upon selecting a reaction
+      selectedAllergicReactions.push({ uuid: event.currentTarget.value });
+      setSelectedAllergicReactions(selectedAllergicReactions);
     } else {
-      patientAlleryReaction.splice(
-        patientAlleryReaction.findIndex(
-          reaction => reaction.uuid == event.target.value
-        ),
-        1
+      // upon deselecting a reaction
+      const modifiedAllergicReactions = selectedAllergicReactions.filter(
+        rxn => rxn.uuid !== event.currentTarget.value
       );
+      setSelectedAllergicReactions(modifiedAllergicReactions);
```

Comment:
It *probably* doesn't matter here, but I think you should avoid the possibility of a race condition by using the functional form of the `set*` function. Something like

```javascript
      setSelectedAllergicReactions(reactions => reactions.push({ uuid: event.currentTarget.value }));
```

and same for the deselect.

Reply:
```javascript
   setSelectedAllergicReactions(reactions =>
     if (event.currentTarget.checked === true) {  // select
       return reactions.push({ uuid: event.currentTarget.value })
     } else { // deselect
       return reactions.filter(rxn => rxn.uuid !== event.currentTarget.value); 
     }
   );
```

You'll probably need to fix the formatting.